internal/cli/feedback/feedback.go

```diff
@@ -0,0 +1,253 @@
+	if formatSelected {
+		panic("output format already selected")
+	}
+	format = f
+	formatSelected = true
+
+	if format == Text {
+		feedbackOut = stdOut
+		feedbackErr = stdErr
+	} else {
+		feedbackOut = bufferOut
+		feedbackErr = bufferErr
+		bufferWarnings = nil
+	}
+}
+
+// GetFormat returns the output format currently set
+func GetFormat() OutputFormat {
+	return format
+}
+
+// Printf behaves like fmt.Printf but writes on the out writer and adds a newline.
+func Printf(format string, v ...interface{}) {
+	Print(fmt.Sprintf(format, v...))
+}
+
+// Print behaves like fmt.Print but writes on the out writer and adds a newline.
+func Print(v string) {
+	fmt.Fprintln(feedbackOut, v)
+}
+
+// Warning outputs a warning message.
+func Warning(msg string) {
+	if format == Text {
+		fmt.Fprintln(feedbackErr, msg)
+	} else {
+		bufferWarnings = append(bufferWarnings, msg)
+	}
+	logrus.Warning(msg)
+}
+
+// FatalError outputs the error and exits with status exitCode.
+func FatalError(err error, exitCode ExitCode) {
+	Fatal(err.Error(), exitCode)
+}
+
+// Fatal outputs the errorMsg and exits with status exitCode.
+func Fatal(errorMsg string, exitCode ExitCode) {
+	if format == Text {
+		fmt.Fprintln(stdErr, errorMsg)
+		os.Exit(int(exitCode))
+	}
+
+	type FatalError struct {
+		Error  string               `json:"error"`
+		Output *OutputStreamsResult `json:"output,omitempty"`
+	}
+	res := FatalError{
+		Error: errorMsg,
+	}
+	if output := getOutputStreamResult(); !output.Empty() {
+		res.Output = output
+	}
+	var d []byte
+	switch format {
+	case JSON:
+		d, _ = json.MarshalIndent(res, "", "  ")
+	case MinifiedJSON:
+		d, _ = json.Marshal(res)
+	case YAML:
+		d, _ = yaml.Marshal(res)
+	default:
+		panic("unknown output format")
+	}
+	fmt.Fprintln(stdErr, string(d))
+	os.Exit(int(exitCode))
+}
+
+func augment(data interface{}) interface{} {
+	if len(bufferWarnings) == 0 {
+		return data
+	}
+	d, err := json.Marshal(data)
+	if err != nil {
+		return data
+	}
+	var res interface{}
+	if err := json.Unmarshal(d, &res); err != nil {
+		return data
+	}
+	if m, ok := res.(map[string]interface{}); ok {
+		m["warnings"] = bufferWarnings
+	}
+	return res
+}
+
+// PrintResult is a convenient wrapper to provide feedback for complex data,
+// where the contents can't be just serialized to JSON but requires more
+// structure.
+func PrintResult(res Result) {
+	var data string
+	switch format {
+	case JSON:
+		d, err := json.MarshalIndent(augment(res.Data()), "", "  ")
+		if err != nil {
+			Fatal(fmt.Sprintf("Error during JSON encoding of the output: %v", err), ErrGeneric)
+		}
+		data = string(d)
+	case MinifiedJSON:
+		d, err := json.Marshal(augment(res.Data()))
+		if err != nil {
+			Fatal(fmt.Sprintf("Error during JSON encoding of the output: %v", err), ErrGeneric)
+		}
+		data = string(d)
+	case YAML:
+		d, err := yaml.Marshal(augment(res.Data()))
+		if err != nil {
+			Fatal(fmt.Sprintf("Error during YAML encoding of the output: %v", err), ErrGeneric)
```

Comment:
```suggestion
			Fatal(tr("Error during JSON encoding of the output: %v", err), ErrGeneric)
		}
		data = string(d)
	case MinifiedJSON:
		d, err := json.Marshal(augment(res.Data()))
		if err != nil {
			Fatal(tr("Error during JSON encoding of the output: %v", err), ErrGeneric)
		}
		data = string(d)
	case YAML:
		d, err := yaml.Marshal(augment(res.Data()))
		if err != nil {
			Fatal(tr("Error during YAML encoding of the output: %v", err), ErrGeneric)
```

Follow established tradition of internationalizing all output strings other than logs and panics.